File: frontend/src/app/shared/models/asset.ts
export interface Asset {
  assetId?: number;               // ✅ match C#
  name: string;
  categoryId: number;
  serialNumber: string;           // ✅ match C# property name
  purchaseDate: string;           // ✅ should be ISO string (YYYY-MM-DD)
  warrantyMonths: number;         // ✅ match C# casing
  statusId: number;
  description: string;
}

File: backend/Models/Asset.cs
using System.Text.Json.Serialization;

namespace backend.Models
{
    public class Asset
    {
        public int AssetId { get; set; }
        public string Name { get; set; }
        public int CategoryId { get; set; }
        public string SerialNumber { get; set; }
        public DateTime PurchaseDate { get; set; }
        public int WarrantyMonths { get; set; }
        public int StatusId { get; set; }
        public string Description { get; set; }

        [JsonIgnore]
        public AssetCategory Category { get; set; }
    }
}
